crates/bevy_render/src/mesh/primitives/dim2.rs

```diff
@@ -266,3 +266,142 @@ impl From<Rectangle> for Mesh {
         rectangle.mesh()
     }
 }
+
+/// A builder used for creating a [`Mesh`] with a [`Capsule2d`] shape.
+#[derive(Clone, Copy, Debug)]
+pub struct Capsule2dMeshBuilder {
+    /// The [`Capsule2d`] shape.
+    pub capsule: Capsule2d,
+    /// The number of vertices used for one hemicircle.
+    /// The total number of vertices for the capsule mesh will be two times the resolution.
+    ///
+    /// The default is `16`.
+    pub resolution: usize,
+}
+
+impl Default for Capsule2dMeshBuilder {
+    fn default() -> Self {
+        Self {
+            capsule: Capsule2d::default(),
+            resolution: 16,
+        }
+    }
+}
+
+impl Capsule2dMeshBuilder {
+    /// Creates a new [`Capsule2dMeshBuilder`] from a given radius, length, and the number of vertices
+    /// used for one hemicircle. The total number of vertices for the capsule mesh will be two times the resolution.
+    #[inline]
+    pub fn new(radius: f32, length: f32, resolution: usize) -> Self {
+        Self {
+            capsule: Capsule2d::new(radius, length),
+            resolution,
+        }
+    }
+
+    /// Sets the number of vertices used for one hemicircle.
+    /// The total number of vertices for the capsule mesh will be two times the resolution.
+    #[inline]
+    pub const fn resolution(mut self, resolution: usize) -> Self {
+        self.resolution = resolution;
+        self
+    }
+
+    /// Builds a [`Mesh`] based on the configuration in `self`.
+    pub fn build(&self) -> Mesh {
+        // The resolution is the number of vertices for one semicircle
+        let resolution = self.resolution as u32;
+        let vertex_count = 2 * self.resolution;
+
+        // Six extra indices for the two triangles between the hemicircles
+        let mut indices = Vec::with_capacity((vertex_count - 2) * 3 + 6);
```

Comment:
Not sure where the error is, but this seems wrong. The default capsule is initially given a capacity of 96, but the length at the end is 90.

Reply:
I believe this should be `(resolution - 2) * 2 * 3 + 6`.

Reply:
Yep, you're correct! I tested with different resolutions to be sure, and that works perfectly